js/main.js

```diff
@@ -0,0 +1,156 @@
+
+const canvas = document.getElementById('wordSearchCanvas');
+const context = canvas.getContext('2d');
+
+
+function redraw(){
+
+  var height = canvas.getBoundingClientRect().height;
+  var width = canvas.getBoundingClientRect().width;
+
+  //console.log('Height is: ' + height + ' Width is: ' + width);
+}
+
+/**
+ * Returns a word seach game board that is created based of the dimensions
+ * of the allocated screen space with each letter being allocated it's
+ * own square within the board.
+ * @param {number} canvasHeight 
+ * @param {number} canvasWidth 
+ * @returns {Object}
+ */
+function createGameBoard(canvasHeight, canvasWidth){
+  var gameBoard = {
+    startPointX: 0,
+    startPointY: 0,
+    gameHeight: 10,
+    gameWidth: 10,
+    board: []
+  }
+
+  var heightScaler = canvasHeight / gameBoard.gameHeight;
+  var widthScaler = canvasWidth / gameBoard.gameWidth;
+
+  if (widthScaler > heightScaler) {
+    widthScaler = heightScaler;
+  } else {
+    heightScaler = widthScaler;
+  }
+
+  var middlePoint = canvasWidth / 2 - (widthScaler * gameBoard.gameWidth/2);
+  gameBoard.startPointX = middlePoint;
+
+  currX = gameBoard.startPointX;
+  currY = gameBoard.startPointY;
+
+  for(var i = 0 ; i < gameBoard.gameHeight ; ++i){
+    var rowOfSquares = [];
+    for(var j = 0 ; j < gameBoard.gameWidth ; ++j){
+
+      var square = {
+        xCoord: currX,
+        yCoord: currY,
+        width: widthScaler,
+        height: heightScaler,
+        character: '-'
+      };
+
+      rowOfSquares.push(square);
+
+      currX += widthScaler;
+      
+    } 
+    gameBoard.board.push(rowOfSquares);
+    currX = gameBoard.startPointX;
+    currY += heightScaler;
+  }
+
+  return gameBoard;
+}
+
+/**
+ * Converts ranNumber from a float to a character in the
+ * english alphabet based on what interval it is within the charFrequency. 
+ * @param {array} charFrequency an array of floats where element at index i 
```

Comment:
Nice work adding the JSDoc.

When you actually run the Closure Compiler over this, I think you will get some type errors. It might be useful to run the compiler soon to find those type errors, even before you ensure the code is compatible with ADVANCED optimizations.

For example, this should be `!Array<number>` instead of just `array`.

Reply:
Good point. I will run the compiler prior to making a PR to try and avoid issues like this going forward. Updated.